_episodes/09-instances.md

```diff
@@ -0,0 +1,276 @@
+```
+In this example, the `.sif` is the image downloaded from Dockerhub, and `mycentos7` is the name that we have
+assigned to the instance. Instead of opening a shell session or executing a command, the container is running on
+the background.
+
+Confirm that the instance is running using the `instance list` command
+```bash
+singularity instance list
+```
+~~~
+INSTANCE NAME    PID      IP    IMAGE
+mycentos7        10782          /home/myuser/centos_centos7.sif
+~~~
+{: .output}
+
+To interact with the instance, the commands `exec` and `shell` available. The instance must be referred as
+`instance://name`.
+For example, to open a shell inside the CentOS instance:
+```bash
+singularity shell instance://mycentos7
+```
+
+Remember that exiting the shell instance will not stop the container. For doing so, use `instance stop`:
+```bash
+singularity instance stop mycentos
+```
+You can confirm the instance doesn't exist with `instance list`.
+
+{% include links.md %}
+
+> ## Instances with bind paths
+>
+> When starting an instance, the same options for bind directories between the host and the container as running
+> an interactive session are available. For example, if you want a directory mounted inside the instance, use the
+> `--bind` option:
+> ```bash
+> singularity instance start --bind /home/user/mydata:/data centos_centos7.sif mycentos7
+> ```
+> binding the directory `mydata/` from the host as `/data` inside the instance.
+{: .callout}
+
+
+## A web server as an instance
+
+One of the main purposes of the Singularity instances is deploying services with customized environments. Before moving
+to more complex use cases, let's start with a basic example: a web service showing a HTML with a message.
+
+Let's write a basic `index.html` file as:
+```html
+<!DOCTYPE html>
+<html>
+<head>
+<title>My awesome service</title>
+</head>
+<body>
+<h1>Hello world!</h1>
+<p>If you see this page, my awesome service is up and running.</p>
+</body>
+</html>
+```
+If you are not familiar with HTML take a quick look at the [HTML Tutorial](https://www.w3schools.com/html/), but it is
+not mandatory. What it really matters is having a minimal webpage that our server will show.
+
+Now, let's prepare a basic web server using [Python http.server](https://docs.python.org/3.9/library/http.server.html).
+Create a definition file, saved as `basicServer.def`, which contains:
+```
+Bootstrap: docker
+From: ubuntu:20.04
+
+%post
+    apt-get update -y
+    apt-get install -y python3.9
+
+%files
+    index.html /tmp/index.html
+
+%startscript
+   cd /tmp
+   python3.9 -m http.server 8850
+```
+If you recall the chapter about [definition files](https://hsf-training.github.io/hsf-training-singularity-webpage/05-definition-files/index.html),
+this definition file will pull the official Ubuntu image from Dockerhub, and will install Python3.9.
+In addition, it copies `index.html` on `/tmp` **inside** the container. When the instance starts, commands specified on
```

Comment:
on -> in ?? (unsure myself)

Reply:
This page says 'in': http://languagelearningbase.com/85099/when-to-use-in-vs-on